admin-dev/themes/default/js/bundle/module/module.js

```diff
@@ -1,1190 +1,841 @@
-                    }
-                });
-            }
-
-            var currentSelector = $(this);
-            var _arrayToSort = arrayToSort;
-            var _currentSelector = currentSelector;
-
-            currentSelector.empty();
-            currentSelector.append('<div class="row">');
-
-            $.each(keysToSort, function(index, value){
-                _currentSelector.find('.row').first().append(_arrayToSort[value].get(0).outerHTML);
-                delete _arrayToSort[value];
-            });
-
-            currentSelector.find('.row').first().append(addonItemHtmlBackup);
-            // Take care of Addons Search Card
-            if ($(moduleItemSelector + ':visible').length != $(moduleItemSelector).length && addonItemHtmlBackup !== null) {
-                $(addonsItemSelector).css('display', 'table');
-            }
-
-            currentSelector.append('</div>');
-        });
-    };
-
-    this.updateTagList = function(tagList) {
-        this.currentTagsList = tagList;
-        /* When this happen we need to update the interface accordingly */
-        this.doSearch();
-    };
-
-    this.resetSearch = function () {
-        // Pick the right selector to process search
-        var moduleItemSelector = this.getModuleItemSelector();
-        var moduleGlobalSelector = this.getModuleGlobalSelector();
-        var _this = this;
-
-        // Reset currentTagsList
-        this.currentTagsList = [];
-        _this.doSearch();
-
-        // Avoid trying to redisplay everything if it's already fully displayed
-        if (this.areAllModuleDisplayed === false) {
-
-            $(moduleGlobalSelector).each(function (index, value) {
-                var totalModules = 0;
-                var _that = _this;
-                $(this).find(moduleItemSelector).each(function (index, value) {
-                    if (_that.currentRefCategory !== null) {
-                        var isFromFilterCategory = ($(this).attr('data-categories') == _that.currentRefCategory);
-                        if (isFromFilterCategory === true) {
-                            totalModules += 1;
-                        }
-                        if ($(this).is(':hidden') && isFromFilterCategory === true) {
-                            $(this).show();
-                        }
-                    } else {
-                        totalModules += 1;
-                        if ($(this).is(':hidden')) {
-                            $(this).show();
-                        }
-                    }
-                });
-
-                // Dont forget this vital var once this done
-                _this.areAllModuleDisplayed = true;
-                _this.updateTotalResults(totalModules, $(this));
-            });
+        if (self.currentTagsList.length) {
+          var tagExists = false;
+          $.each(self.currentTagsList, function(index, value) {
+            value = value.toLowerCase();
+            tagExists |= (
```

Comment:
I want to show a module if one of all the tag exists on any value.

As we work with booleans, the binary operator work. I'll try to explain:

By default the `tagExists` value is `false`.
I don't wan't this value to be `false` if already to `true`. This is a binary `OR` operator, so:

```
0 |= 0 === 0
1 |= 0 === 1
0 |= 1 === 1
1 |= 1 === 1
```

In this case: if a value of a tag is found the value is set to 1. For every next tag, if the value cannot be found, the value will stay to 1.